powdr/src/lib.rs

```diff
@@ -64,25 +65,53 @@ impl Session {
     }
 
     pub fn prove(&mut self) {
-        match self.pipeline.read_constants_mut(Path::new(&self.out_path)) {
-            Ok(_) => (),
-            Err(_) => {
-                self.pipeline.compute_fixed_cols().unwrap();
+        let asm_name = self.pipeline.asm_string().unwrap().0.clone().unwrap();
+        let pil_file = create_pil_file_path(&asm_name);
+
+        let generate_artifacts = if let Some(existing_pil) = read_file_if_exists(&pil_file) {
```

Comment:
i think the `if_exists` can be removed also, you can't read the file if it doesn't exist :P